src-tauri/src/plugin_management/download.rs

```diff
@@ -0,0 +1,162 @@
+use app::GameInstall;
+use once_cell::sync::OnceCell;
+use std::{
+    fs::{self, File, OpenOptions},
+    io,
+    path::PathBuf,
+    str::FromStr,
+};
+use tauri::{
+    async_runtime::{block_on, channel, Mutex, Receiver, Sender},
+    Manager, State,
+};
+use thermite::{core::utils::TempDir, prelude::ThermiteError};
+use zip::ZipArchive;
+
+use crate::{mod_management::ParsedThunderstoreModString, APP_HANDLE};
+
+static INSTALL_STATUS_RECV: OnceCell<Mutex<Receiver<bool>>> = OnceCell::new();
+
+pub struct InstallStatusSender(Mutex<Sender<bool>>);
+
+impl InstallStatusSender {
+    pub fn new() -> Self {
+        let (send, recv) = channel(1);
+
+        INSTALL_STATUS_RECV
+            .set(Mutex::new(recv))
+            .expect("failed to set INSTALL_STATUS_RECV");
+
+        Self(Mutex::new(send))
+    }
+}
+
+/// Tries to install plugins from a thunderstore zip
+pub async fn install_plugin(
+    game_install: &GameInstall,
+    zip_file: &File,
+    thunderstore_mod_string: &str,
+    can_install_plugins: bool,
+) -> Result<(), ThermiteError> {
+    let plugins_directory = PathBuf::new()
+        .join(&game_install.game_path)
+        .join("R2Northstar")
+        .join("plugins");
+    let temp_dir = TempDir::create(plugins_directory.join("___flightcore-temp-plugin-dir"))?;
+    let manifest_path = temp_dir.join("manifest.json");
+    let mut archive = ZipArchive::new(zip_file)?;
+
+    let parsed_mod_string = ParsedThunderstoreModString::from_str(thunderstore_mod_string).map_err(|_|ThermiteError::MiscError("Gecko why is this returning nothing? anyway the error is failed to parse thunderstore string lmao".into()))?;
```

Comment:
I assume this will be changed with #295 merged? :P

Reply:
yes